spring-grpc-core/src/test/java/org/springframework/grpc/server/service/DefaultGrpcServiceDiscovererTests.java

```diff
@@ -43,28 +43,55 @@
  */
 class DefaultGrpcServiceDiscovererTests {
 
+	@Test
+	void whenNoServicesRegisteredThenListServiceNamesReturnsEmptyList() {
+		new ApplicationContextRunner().withUserConfiguration(DefaultGrpcServiceDiscovererTestsBaseConfig.class)
+			.run((context) -> assertThat(context).getBean(DefaultGrpcServiceDiscoverer.class)
+				.extracting(DefaultGrpcServiceDiscoverer::listServiceNames, InstanceOfAssertFactories.LIST)
+				.isEmpty());
+	}
+
+	@Test
+	void whenServicesRegisteredThenListServiceNamesReturnsNames() {
+		new ApplicationContextRunner()
+			.withUserConfiguration(DefaultGrpcServiceDiscovererTestsBaseConfig.class,
+					DefaultGrpcServiceDiscovererTestsServiceConfig.class)
+			.run((context) -> assertThat(context).getBean(DefaultGrpcServiceDiscoverer.class)
+				.extracting(DefaultGrpcServiceDiscoverer::listServiceNames, InstanceOfAssertFactories.LIST)
+				.containsExactly("serviceB", "serviceA"));
+
+	}
+
 	@Test
 	void servicesAreFoundInProperOrderWithExpectedGrpcServiceAnnotations() {
-		new ApplicationContextRunner().withUserConfiguration(DefaultGrpcServiceDiscovererTestsConfig.class)
-			.run((context) -> {
-				assertThat(context).getBean(DefaultGrpcServiceDiscoverer.class)
-					.extracting(DefaultGrpcServiceDiscoverer::findServices, InstanceOfAssertFactories.LIST)
-					.containsExactly(DefaultGrpcServiceDiscovererTestsConfig.SERVICE_DEF_B,
-							DefaultGrpcServiceDiscovererTestsConfig.SERVICE_DEF_A);
-				TestServiceConfigurer configurer = context.getBean(TestServiceConfigurer.class);
-				assertThat(configurer.invocations).hasSize(2);
-				assertThat(configurer.invocations.keySet()).containsExactly(SERVICE_B, SERVICE_A);
-				assertThat(configurer.invocations).containsEntry(SERVICE_B, null);
-				assertThat(configurer.invocations).hasEntrySatisfying(SERVICE_A, (serviceInfo) -> {
-					assertThat(serviceInfo.interceptors()).isEmpty();
-					assertThat(serviceInfo.interceptorNames()).isEmpty();
-					assertThat(serviceInfo.blendWithGlobalInterceptors()).isFalse();
-				});
```

Comment:
This is one of the fruits of this simplification (goodbye verifying cofigurer things in discoverer tests)